envoy/common/regex.h

```diff
@@ -22,6 +22,12 @@ class CompiledMatcher : public Matchers::StringMatcher {
    */
   virtual std::string replaceAll(absl::string_view value,
                                  absl::string_view substitution) const PURE;
+
+  /**
+   * Returns a string representation of the Regex matcher (the pattern to be
+   * matched).
+   */
+  virtual const std::string& stringRepresentation() const PURE;
```

Comment:
This would be slightly clearer to me if it were named `pattern()` or `regex()`, but I don't feel strongly about it

Reply:
I agree it would be a better name. Changing to `pattern()`.